domainbed/networks.py

```diff
@@ -210,3 +211,24 @@ def Classifier(in_features, out_features, is_nonlinear=False):
             torch.nn.Linear(in_features // 4, out_features))
     else:
         return torch.nn.Linear(in_features, out_features)
+
+
+class Whole(nn.Module):
```

Comment:
Why is this class necessary? Wouldn't a `nn.Sequential` declared inside `Fish` incorporate all these methods?

Reply:
Hi David! Thanks for the quick response. This class is necessary because we need the `reset_weights` method during the training of Fish to apply new weights to the model. It also makes creating a clone of the meta-model without copying the gradient much easier.

If you think this is too clunky I can try to work around it and encapsulate as much into the Fish class as possible.

Reply:
Sounds good! Could we call it `WholeFish` or something like that? Thanks!

Reply:
Ok! Just to double-check, I can leave the class as it is and rename it as `WholeFish`?

Reply:
Yes, we can leave it like that. I will look into sub-classing for Fish and ARM in the near future.